packages/react-components/src/components/ChatMessage/ChatMessageContent.tsx

```diff
@@ -152,3 +160,63 @@ const messageContentAriaText = (props: ChatMessageContentProps): string | undefi
         })
     : undefined;
 };
+
+type NodeProcessInstruction = {
+  shouldProcessNode: unknown;
+  processNode: unknown;
+};
+
+const processHtmlToReact = (props: ChatMessageContentProps): JSX.Element => {
+  const htmlToReactParser = new Parser();
+
+  /* @conditional-compile-remove(teams-inline-images) */
+  const processInlineImage: NodeProcessInstruction = {
+    // Custom <img> processing
+    shouldProcessNode: (node): boolean => {
+      // Process img node with id in attachments list
+      return (
+        node.name &&
+        node.name === 'img' &&
+        node.attribs &&
+        node.attribs.id &&
+        props.message.attachedFilesMetadata?.find((f) => f.id === node.attribs.id)
+      );
+    },
+    processNode: (node, children, index): HTMLElement => {
+      // logic to check id in map/list
+      const fileMetadata = props.message.attachedFilesMetadata?.find((f) => f.id === node.attribs.id);
+      // if in cache, early return
+      if (props.attachmentsMap && node.attribs.id in props.attachmentsMap) {
+        node.attribs = { ...node.attribs, src: props.attachmentsMap[node.attribs.id] };
+        return processNodeDefinitions.processDefaultNode(node, children, index);
+      }
+      // not yet in cache
+      if (fileMetadata && props.onFetchAttachment && props.attachmentsMap) {
+        props.onFetchAttachment(fileMetadata);
```

Comment:
We design this so it doesn't need to await and can render immediately (even though it might be broken image), and when the URl gets updated in the attachmentsMap, will trigger a re-render with the proper image src